src/test/java/com/melon/portfoliomanager/services/AnalyticsServiceTest.java

```diff
@@ -0,0 +1,188 @@
+package com.melon.portfoliomanager.services;
+
+import com.melon.portfoliomanager.dtos.responses.AnalyticsResponse;
+import com.melon.portfoliomanager.dtos.responses.AssetStat;
+import com.melon.portfoliomanager.dtos.responses.StockPricesDTO;
+import com.melon.portfoliomanager.exceptions.NoSuchUserException;
+import com.melon.portfoliomanager.models.PortfolioItem;
+import com.melon.portfoliomanager.models.User;
+import com.melon.portfoliomanager.repositories.PortfolioItemRepository;
+import com.melon.portfoliomanager.repositories.TransactionRepository;
+import com.melon.portfoliomanager.repositories.UserRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.system.OutputCaptureExtension;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ExtendWith(OutputCaptureExtension.class)
+public class AnalyticsServiceTest {
+
+    @MockBean
+    private TransactionRepository transactionRepository;
```

Comment:
Remove this dependency - it is not used anywhere.

Reply:
fixed